### PR TITLE
Add memory testing

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
Uses valgrind if it is found to test the unit tests to ensure there are no intrinsic memory errors. @junghans how do we ensure that valgrind is installed on the gitlab runners and travis I did not see where you were configuring cmake and other dependencies in the .ci-.. files?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,7 +42,12 @@ include(GNUInstallDirs)
 
 option(ENABLE_TESTING "Build and enable testing stuff" OFF)
 if(ENABLE_TESTING)
+  include(CTest)
```
Comment (junghans) [inline CMakeLists.txt]: Why do we need this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -42,7 +42,12 @@ include(GNUInstallDirs)
...
   enable_testing()
+  find_program(VALGRIND "valgrind")
+  if(NOT VALGRIND)
```
Comment (junghans) [inline CMakeLists.txt]:
```suggestion
  find_program(VALGRIND_EXECUTABLE "valgrind")
  find_package_handle_standard_args(VALGRIND REQUIRED_VARS VALGRIND_EXECUTABLE)
```
CMake default message is better!

File: src/tests/CMakeLists.txt [hunk 3]
```diff
@@ -38,4 +38,7 @@ foreach(PROG
   # run tests for tools (for coverage) as well
   set_tests_properties(unit_${PROG} PROPERTIES LABELS "tools;votca")
 
+  if(VALGRIND)
+    add_test(NAME memory_${PROG} COMMAND "${VALGRIND}" --tool=memcheck --leak-check=yes --show-reachable=yes --num-callers=20 --track-fds=yes --error-exitcode=1 $<TARGET_FILE:unit_${PROG}>)
```
Comment (junghans) [inline src/tests/CMakeLists.txt]: Please label the new test as well.
```suggestion
    add_test(NAME memory_${PROG} COMMAND "${VALGRIND}" --tool=memcheck --leak-check=yes --show-reachable=yes --num-callers=20 --track-fds=yes --error-exitcode=1 $<TARGET_FILE:unit_${PROG}>)
    set_tests_properties(memory_${PROG} PROPERTIES LABELS "tools;votca")
```

File: CMakeLists.txt (hunk 2 above)
Comment (JoshuaSBrown) [reply CMakeLists.txt]: Yes, but that command is only supported in the most recent versions of cmake, I have cmake 3.10 and it is not recognized.

File: CMakeLists.txt (hunk 1 above)
Comment (JoshuaSBrown) [reply CMakeLists.txt]: I guess not.

File: CMakeLists.txt (hunk 2 above)
Comment (JoshuaSBrown) [reply CMakeLists.txt]: Never mind, I was missing an include.

File: src/tests/CMakeLists.txt [hunk 4]
```diff
@@ -38,4 +38,8 @@ foreach(PROG
   # run tests for tools (for coverage) as well
   set_tests_properties(unit_${PROG} PROPERTIES LABELS "tools;votca")
 
+  if(VALGRIND_EXECUTABLE)
```
Comment (junghans) [inline src/tests/CMakeLists.txt]:
```suggestion
  if(VALGRIND_FOUND)
```
`find_package_handle` defines that.

**Pull request merged**